lib/pbench/test/unit/server/test_datasets_list.py

```diff
@@ -47,6 +48,43 @@ class TestDatasetsList:
     fixture.
     """
 
+    @pytest.fixture()
+    def aliases(self):
+        mtable = aliased(Metadata)
+        stable = aliased(Metadata)
+        gtable = aliased(Metadata)
+        utable = aliased(Metadata)
+        aliases = {
+            Metadata.METALOG: mtable,
+            Metadata.SERVER: stable,
+            Metadata.GLOBAL: gtable,
+            Metadata.USER: utable,
+        }
+        query = (
+            Database.db_session.query(Dataset)
+            .outerjoin(
+                mtable,
+                and_(mtable.dataset_ref == Dataset.id, mtable.key == Metadata.METALOG),
+            )
+            .outerjoin(
+                stable,
+                and_(stable.dataset_ref == Dataset.id, stable.key == Metadata.SERVER),
+            )
+            .outerjoin(
+                gtable,
+                and_(gtable.dataset_ref == Dataset.id, gtable.key == Metadata.GLOBAL),
+            )
+            .outerjoin(
+                utable,
+                and_(
+                    utable.dataset_ref == Dataset.id,
+                    utable.key == Metadata.USER,
+                    utable.user_id == DRB_USER_ID,
+                ),
```

Comment:
in the source code you are checking the `auth_id` before doing an outer join on `utable` should we be using that conditional here as well?

Reply:
The test setup is a bit more fragile since we're installing a `get_auth_id` *mock* within the test, after the fixture runs. I could probably have made this smarter, but the tests I was concerned with all use `drb` and it didn't seem worthwhile. But a comment on the fixture might have been a good idea. 😦